Automattic/jetpack
# VideoPress: use "Processing" thumbnail placeholder while a new thumbnail is being saved

Part of #39524.

## Proposed changes:

* The thumbnail update is a process that depends on the backend generating the new image file and updating the video information
* This may take some time, and while the image is being processed, the video details edit page inside VideoPress dashboard was showing the old thumbnail after saving the changes, what could be misleading to users (and result on people refreshing the page and getting a broken, 404 thumbnail)
* We have, on the local state, a prop to control the poster updating status, so this PR is using it to set a "Processing" thumbnail placeholder while the image is not ready on the backend

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Make sure you have a Jetpack site with a VideoPress license and with the VideoPress standalone plugin installed, so you have access to the VideoPress Dashboard
* Go to Jetpack > VideoPress and upload a new file
* After the upload is ready, click the "Edit video details" button
* On the video thumbnail preview, click the pencil button to edit the thumbnail and then click "Select from video"
* Choose a different thumbnail and click "Select this frame"
* Click "Save changes"
* Confirm the video thumb component will show a processing message while the details are saved
* Confirm the video thumb keeps showing a processing message even after the "save" action is done, because the new video thumb is still being processed
* Confirm the new video thumb will eventually load properly

Video showing what to expect:

https://github.com/user-attachments/assets/b0cce43f-3295-4700-b6ab-67c7a86d26e6
